iohk/UpdateProposal.hs

```diff
@@ -113,14 +115,19 @@ parseUpdateProposalCommand = subparser $
               <> help "Date string identifying the update proposal"
 
     updateProposalConfig :: Parser UpdateProposalConfig1
-    updateProposalConfig = UpdateProposalConfig1
-      <$> ( option (eitherReader (gitRevision . T.pack))
-            (long "revision" <> short 'r' <> metavar "SHA1"
-              <> help "Daedalus revision to fetch") )
-      <*> ( fmap T.pack $ strOption (long "block-version" <> short 'B' <> metavar "VERSION"
-           <> help "Last known block version. Check the wiki for more info.") )
-      <*> ( option auto (long "voter-index" <> short 'V' <> metavar "INTEGER"
-           <> help "A number representing you, the vote proposer. Check the wiki for more info.") )
+    updateProposalConfig = UpdateProposalConfig1 <$> revisionP <*> blockVersionP <*> voterIndexP <*> releaseNotesP
+
+    revisionP :: Parser GitRevision
+    revisionP = option (eitherReader (gitRevision . T.pack)) (long "revision" <> short 'r' <> metavar "SHA1" <> help "Daedalus revision to fetch")
+
+    blockVersionP :: Parser Text
+    blockVersionP = fmap T.pack $ strOption (long "block-version" <> short 'B' <> metavar "VERSION" <> help "Last known block version. Check the wiki for more info.")
+
+    voterIndexP :: Parser Int
+    voterIndexP = option auto (long "voter-index" <> short 'V' <> metavar "INTEGER" <> help "A number representing you, the vote proposer. Check the wiki for more info.")
+
+    releaseNotesP :: Parser (Maybe Text)
+    releaseNotesP = (Just . T.pack <$> strOption ( long "release-notes" <> metavar "RELEASE_NOTES" <> (help "Path to release notes (html)") <> (completer (bashCompleter "file") )))
```

Comment:
Better use the `FilePath` for all paths and `FP.fromText <$> option auto ...` to parse the option.

Reply:
the problem, is that i need the parser to return a `FilePath`, but then i want to store the `Text` within that file in the data record
and then i need 2 data records, for the parser, and the result after applying IO

Reply:
OK fair enough